lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -62,10 +79,264 @@ public function init() {
 		}
 		add_action( $hook, array( $this, 'generate_and_enqueue_styles' ) );
 
+		add_action( 'init', array( $this, 'register_filter_for_current_template_webfonts_enqueuing' ) );
+		add_action( 'init', array( $this, 'enqueue_webfonts_used_in_global_styles' ) );
+
+		add_filter( 'the_content', array( $this, 'enqueue_webfonts_used_in_content' ) );
+
+		$this->register_webfont_cache_invalidation_actions();
+
 		// Enqueue webfonts in the block editor.
 		add_action( 'admin_init', array( $this, 'generate_and_enqueue_editor_styles' ) );
 	}
 
+	/**
+	 * Enqueue the webfonts used in the content.
+	 *
+	 * @param string $content The post content.
+	 *
+	 * @return string
+	 */
+	public function enqueue_webfonts_used_in_content( $content ) {
+		global $post;
+
+		$webfonts_used_in_post_cache = get_post_meta( $post->ID, self::$webfonts_cache_meta_attribute, true );
+
+		if ( ! $webfonts_used_in_post_cache ) {
+			$webfonts_used_in_post_cache = $this->collect_fonts_from_block_markup( $post->post_content );
+
+			if ( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) {
+				update_post_meta( $post->ID, self::$webfonts_cache_meta_attribute, $webfonts_used_in_post_cache );
+			}
+		}
+
+		$webfont_ids = array_keys( $webfonts_used_in_post_cache );
+
+		foreach ( $webfont_ids as $webfont ) {
+			if ( isset( $this->get_registered_fonts()[ $webfont ] ) ) {
+				$this->enqueue_font( $webfont );
+			}
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Set up the actions to invalidate webfont cache.
+	 */
+	private function register_webfont_cache_invalidation_actions() {
```

Comment:
In this function, I thought of two approaches:

1. Hook into `save_post` and get the $post->post_type to determine the invalidation method (theme mod or meta attribute);
2. The one I picked. I feel it's easier to understand and it looks to fit more the WP filter/action definitions.